katsdpdata/prod_handler.py

```diff
@@ -406,6 +408,17 @@ def metadata_transfer_complete(self):
         met = mh.set_product_received(met)
         met_bucket = self.get_bucket_stats()
         mh.add_bucket_stats(met, met_bucket)
+        self.solr = pysolr.Solr(self.solr_url)
+        for product_type in ['MeerKATVisibilityProduct', 'MeerKATFlagProduct']:
+            query = 'CaptureBlockId:{} AND CAS.ProductTypeName:{}'.format(met['id'],product_type)
```

Comment:
You should use met['CaptureBlockId'], not met['id']. The id might not always be a CaptureBlockId.

You could also make your query

```
query = 'CaptureBlockId:{} AND CAS.ProductTypeName:MeerKATVisibilityProduct OR CAS.ProductTypeName:MeerKATFlagProduct'.format(met['CaptureBlockId'])
```

Then you can loop through the results. No point doing 2 queries to solr when you can do one query.